1. Reverse a given number loops and equations and return true if it is the same as the original number.

In [ ]:
n= int(input("enter Num : " ))

rev=int(0)

last = n%10
while n!=0:
    rev=rev*10+last
    n//= 10
    last = int (n%10)  
print(rev)    

    

2. Return the number of times that the string “Emma” appears anywhere in the given string. 

In [ ]:
n = input("enter any string :")
print(n.count("Emma"))

3. Print the following pattern. 
1 
2 2 
3 3 3 
4 4 4 4 
5 5 5 5 5   


In [ ]:
for i in range (1,6):
    print(str(i)*i)

4. Given a string of odd length greater than 7, return a string made of the middle three chars of a given String.

In [ ]:
s = input(" Enter String : " )

l = int(len(s))

l=l//2
print(str (s [l-1:l+2] ))

5. Given 2 strings, s1 and s2, create a new string by appending s2 in the middle of s1.

In [ ]:
s1 = input(" Enter String : " )
s2 = input(" Enter String : " )

l1 = len(s1)
l1 = l1//2

h = s1[:l1:]+s2+s1[l1:]
 
print(h)


6. Given 2 strings, s1, and s2 return a new string made of the first, middle and last char each input string.

In [ ]:
s1 = input(" Enter String : " )
s2 = input(" Enter String : " )

l1 = len(s1)
l1 = l1//2

l2= len(s2)
l2= l2//2

news1 = s1[0] + s1[l1] + s1[len(s1)-1]
news2 = s2[0] + s2[l2] + s2[len(s2)-1]

news3= news1+news2

print(news3)


7. arrange String characters such that lowercase letters should come first

In [ ]:
s1 = input(" Enter String : " )

svar=""
cvar=""
for i in s1:
    if i  == i.lower():
       
        svar+=i
    else:
        cvar+=i
        
all1 = svar+cvar  

print(all1)      



8. Sort a given string alphabetically. 

In [ ]:
a="zxy"
b = ''.join(sorted(a))
print(b)

##### 9. Given a string input Count all lower case, upper case, digits, and special symbols Sample Input: ”My?Name!Is956Yazeed” Sample Output: Number of upper case: 4 Number of lower case: 10 Number of digits: 3 Number of special symbols: 2 

In [ ]:
x = input( "Enter any string please :")
uc=0
lc=0
digit=0
s=0
for i in x:
    if i.isalpha() and i==i.upper():
        uc+=1
    elif i.isalpha() and i==i.lower():
        lc+=1
    elif i.isnumeric():
        digit+=1
    else:
        s+=1
        
print(uc)
print(lc)
print(digit)
print(s)



10. Write c program that takes a positive integer n as an input from the user and calculates the sum up to n.

In [ ]:
#rule : (n)(n + 1)/2
n = int(input("Enter a positive integer: "))
sum = n * (n+1) / 2
print(sum)

11. Use the While or for statement to write a program that computes and displays the factorial of a positive integer n entered by the user.

In [ ]:
n= int(input("enter number : to calculate fact : "))
for  i  in range (4,0,-1):
    n*=i
print(n)




12. Write a program that uses a while loop to display valuesfrom -5 through 5, using an increment of 0.5.

In [ ]:
c = -5.0
while  c<=5:
    print(c)
    c+=0.5

13. Write a c program that prints the following star pattern:
* * * *
* * *
* * 
*

In [ ]:
r=4
for i in range (4):
    print("*"*r)
    r=r-1

14. Write a c program that prints the following star pattern:
*
* *
* * *
* * * *
* * * * *
* * * *
* * *
* *
*

In [ ]:
for i in range (6):
    print('*'*i)
r=4
for i in range (4):
    print("*"*r)
    r=r-1

15. Write a c program that prints the following star pattern
7
* 6
* * 5 
* * * 4 
* * * * 3


In [ ]:

n=7
for i in range (5):
    print('*'*i,n)
    n=n-1


16. Write a c program that prints the following star pattern:
    *
   * *
  * * *
 * * * *
* * * * *

In [ ]:
num  = 5

for i in range (0,num):
    for j in range (0,num-i-1):
        print(end=" ")
    for j in range (0,i+1):
        print('*',end=" ")
    print()

17. Write a c program that prints the following star pattern:
* * * * *
 * * * *
  * * *
   * *
    *

In [ ]:
num  = 5

for i in range (num,0,-1):
    for j in range (0,num-i):
        print(end=" ")
    for j in range (0,i):
        print('*',end=" ")
    print()

18. Write a c program that prints the following star pattern:
    * 
   * * 
  * * * 
 * * * * 
* * * * * 
 * * * * 
  * * * 
   * * 
    * 

In [ ]:
num  = 5


for i in range (0,num):
    for j in range (0,num-i-1):
        print(end=" ")
    for j in range (0,i+1):
        print('*',end=" ")
    print()
    
    
    num  = 5
    

for i in range (num,0,-1):
    for j in range (0,num-i):
        print(end=" ")
    for j in range (0,i):
        print('*',end=" ")
    print()

19. Write programs that implement the following equations:

1 -  series n^2 from 1 to 100
2 -  series n^2 * (n-5) from 6 tp 100

In [ ]:
sum1=0.0

for i in range (1,101):
    sum+=i**2
print (sum1)

sum2=0.0

for i in range (6,101):
    sum2+= i**2 * (i-5)

print (sum2)
    

20. Write a Python program to print Fibonacci series up to 10th term. 

In [ ]:
nterms = 10
n1 = 0
n2 = 1
count = 0
while count < 10:
       print(n1)
       nth = n1 + n2
       n1 = n2
       n2 = nth
       count += 1

21. Write a Python program to print GCD between any two given numbers.

In [ ]:
print (" enter the small number then the large number ")

x = int( input("enter x : " ))
y = int( input("enter y : " ))

while(y):
       x, y = y, x % y

print(x)

22. Write a Python program to print LCM between any two given numbers. 

In [ ]:
 
print (" enter the small number then the large number ")

x = int( input("enter x : " ))
y = int( input("enter y : " ))

gcd = 0
while(y):
       x, y = y, x % y
gcd=x
        
print ( a*b/gcd)

23. Write a Python program to find the sum of all natural odd numbers between 1 and 100. 

In [ ]:
for i in range(1,101):
    if(i%2!=0):
        print (i , end = '  ')
        

24. Write a Python program to find the sum of all natural prime numbers between 1 and 100. 

In [ ]:

sum=0
for num in range(101):  
   if num > 1:  
       for i in range(2,num):  
           if (num % i) == 0:  
               break  
       else:  
           sum+=num
               
print(sum)

25. Write a Python program to find the product of all natural even numbers between 1 and 100

In [ ]:

sum=0
for num in range(101):  
   if num > 1:  
       for i in range(2,num):  
           if (num % i) == 0:  
               break  
       else:
           if num%2==0
           sum+= num * num
               
print(sum)

26. Write a Python program that prints the multiplication tables of all numbers between 0 and 9. 

In [ ]:
for l in range(0,10):
    for i in range(1,11):
        print(l,'x',i,'=',l*i)

27. Write a Python program to input a number from user and count number of digits in the given integer.

In [ ]:
n = int(input("enter number : "))

c=0

while n!=0:
    c=c+1
    n=n//10
    

print(c)

28. Write a Python program to input a number from user and find if the number is Palindrome. 

In [2]:
num = input(' Enter any number : ')
val = int(num)
if num == str(num)[::-1]:
    print('The given number is PALINDROME')
else:
    print('The given number is NOT a palindrome')

 Enter any number : 1124425
The given number is NOT a palindrome


29. Write a Python program to input a number and calculate the sum of its digits. 

In [3]:
n = int(input("enter number : "))
sum=0
while n!=0:
    sum+= n%10
    n=n//10
    
print(sum)

enter number : 111
3


30. Write a Python program find the maximum number to get of 3-3 digit palindrome(do not use strings).

In [4]:
#ambiguous

31. Given a string, return the sum and average of the digits that appear in the string, ignoring all other characters. 

In [5]:
n = input("enter number : ")
sum=0
c =0
for i in n:
    if(i.isnumeric()):
        sum+= int(i)
        c+=1
        
print("sum : "  , sum)
print("avg : " , sum/c)

    

enter number : 123
sum :  6
avg :  2.0


32. Find all occurrences of “USA” in given string ignoring the case

In [6]:
s = input("enter string : ")

temp = s.upper()
print ( " number of occurrences " ,temp.count("USA") )

enter string : usa usa usa usa usa UsA
 number of occurrences  6


33. ApplytheStringcharactersbalanceTest(occurrencesofallappearedcharactersmustbeequal). 

In [7]:
s1 = input("enter string 1 : ")
s2 = input("enter string 2 : ")
flag = True
for char in s1:
    if char in s2:
        continue
    else:
        flag = False
        
print(flag)

enter string 1 : a
enter string 2 : b
False


34. Given a string and and int n, remove n characters from string by each time until the string has no more characters.

In [8]:
s1 = input("enter string 1 : ")
n =int(input("enter num "))


for i in s1 :
    if(i==str(n)):
       s2 =  s1.replace(str(n),"")
        
print(s2)
        
    

enter string 1 : ahmad12324
enter num 2
ahmad134


35. Accept string from the user and display only those characters which are present at an even index 

In [9]:
s1 = input("enter string 1 : ")

for i in range (1,len(s1),2):
    print(s1[i])
    

    
        

        
    

enter string 1 : 123456
2
4
6


36. Remove duplicate from a string and print the results.

In [ ]:
s1 = input("enter string :")

for  i in range(len(s1)) :
    for j in range(1,len(s1)):
        if(s1[i] is s1[j]):
          s2 =  s1.replace(s1[j],"")
print(s2)

37. Remove duplicate from a string and print how much time each character appeared. 

In [ ]:
from collections import OrderedDict
str1 = " ahmad"
print ("".join(OrderedDict.fromkeys(str1)) )

38. Find all unique character, numbers, ...etc. in a string and print them.

In [ ]:
x = input( "Enter any string please :")
for i in x:
    if i.isalpha()==False:
        print(i , end = " ")
   


39. Write a Python program to accept a filename from the user as a string and print the extension of that file. Sample filename : abc.java Output : java

In [ ]:
f = input(" enter  the Filename: ")
f_extns = f.split(".")
print ("The extension of the file is : " + f_extns[-1])

40. Write a Python program to get a new string from a given string where ”Is” has been added to the front. If the given string already begins with ”Is” then return the string unchanged. 

In [ ]:
s = input(" enter  s: ")
if len(s) >= 2 and s[:2] == "Is":
    print(s)
else:
    print("Is" + s)

41. Write a Python program to get a string which is n (non-negative integer) copies of a given string

In [ ]:
n = int(input ("enter n :"))
s = input("enter strin ")

print (s*n)

42. Write a Python program to get the n (non-negative integer) copies of the first 2 characters of a given string. Return the n copies of the whole string if the length is less than 2.

In [ ]:
n = int(input ("enter n :"))
str = input("enter strin ")

flen = 2
if flen > len(str):
    flen = len(str)
substr = str[:flen]
  
result = ""
for i in range(n):
    result = result + substr
    
print(result)

43. WriteaPythonprogramtoprintoutthesetofallthecharactersfromSt1whicharenotpresent in St2 

In [ ]:

str1 = input("enter str1 :")
str2 = input("enter str2 :")
s3=""
for  i in str1:
    for j in str2:
        if(i!=j):
          s3+=i
            
print(s3)      



44. Write a Python program to display your given details like name, age, address in three different lines.

In [ ]:
name = "ahmad"
age = 24
address = " irbid - jordan "
print("Name: {}\nAge: {}\nAddress: {}".format(name, age, address))

45. Write a Python program to sort three integers without using conditional statements, loops or strings.

In [ ]:
x = int(input("Input first number: "))
y = int(input("Input second number: "))
z = int(input("Input third number: "))

a1 = min(x, y, z)
a3 = max(x, y, z)
a2 = (x + y + z) - a1 - a3
print("Numbers in sorted order: ", a1, a2, a3)

In [ ]:
46. WriteaPythonprogramtogetthesummationoftheASCIIvaluesofaaacharactersinastring. 

In [ ]:
str1 = input("enter str1 :")

sum = 0
 
for i in str1:
     sum+=ord(i)

print(sum)

In [ ]:
47. Write a Python program to convert a byte string to a list of integers. 

In [ ]:


x = b"string"
print()
print(list(x))
print()

In [ ]:
48. Write a Python program to get the identity of an object. 

In [ ]:
str1  = "aaa"
obj1_address = id(str1)
print(obj1_address)


In [ ]:
49. Write a Python program to get the system time.

In [ ]:
import datetime
time = datetime.datetime.now()
print (time)

In [ ]:
50. Write a Python program to find the ASCII string of a given number. 

In [ ]:
n = int(input("enter n :"))

print(chr(n))

51. Write a Python program to prove that two string variables of same value point same memory location. 

In [ ]:
n="a"
x="a"

if x is n :
    print("true")

52. Write a Python function that takes a positive integer and returns the sum of the cube of all the positive integers smaller than the specified number.

In [ ]:
n = int(input("enter n :"))
n -= 1
total = 0
while n > 0:
    total += n * n * n
    n -= 1
print (total)

53. Write a Python program to add two positive integers without using the ’+’ operator. Note: Use bit wise operations to add two numbers 

In [ ]:
a = int(input("enter a :"))
b = int(input("enter b :"))
while b != 0:
        data = a & b
        a = a ^ b
        b = data << 1
print (a)

54. Write a Python program to get all strobogrammatic numbers that are of length n. A strobogrammatic number is a number whose numeral is rotationally symmetric, so that it appears the same when rotated 180 degrees. In other words, the numeral looks the same rightside up and upside down (e.g., 69, 96, 1001). For example, Given n = 2, return [”11”, ”69”, ”88”, ”96”]. Given n = 3, return [’818’, ’111’, ’916’, ’619’, ’808’, ’101’, ’906’, ’609’, ’888’, ’181’, ’986’, ’689’]

In [ ]:
# using google .. # and im still not understanding what is this ..
#https://github.com/keon/algorithms/blob/master/math/generate_strobogrammtic.py
def gen_strobogrammatic(n):
    """
    :type n: int
    :rtype: List[str]
    """
    result = helper(n, n)
    return result


def helper(n, length):
    if n == 0:
        return [""]
    if n == 1:
        return ["1", "0", "8"]
    middles = helper(n-2, length)
    result = []
    for middle in middles:
        if n != length:
            result.append("0" + middle + "0")
        result.append("8" + middle + "8")
        result.append("1" + middle + "1")
        result.append("9" + middle + "6")
        result.append("6" + middle + "9")
    return result

print("n = 2: \n",gen_strobogrammatic(2))
print("n = 3: \n",gen_strobogrammatic(3))
print("n = 4: \n",gen_strobogrammatic(4))


55. Write a Python program to find the median among n given numbers. 

In [ ]:
# n nuumbers ? in list or in  var or what please complain ..

56. Write a Python program to find all the divisors of a given number. 

In [ ]:
n=int(input("Enter an number:"))
print("DIVISORS :")
for i in range(1,n+1):
    if(n%i==0):
        print(i)

57. Write a Python program to find the number of divisors of a given integer is even or odd.

In [ ]:
n=int(input("Enter an number:"))
for i in range(n):
    x = len([i for i in range(1,n+1) if not n % i])
print( x)

58. Write a Python program to create a sequence where the first four members of the sequence are equal to one, and each successive term of the sequence is equal to the sum of the four previous ones. Find the Nth member of the sequence.


In [ ]:
# i need more complain

59. WriteaPythonprogramthatacceptapositivenumberandsubtractfromthisnumberthesum of its digits and so on. Continues this operation until the number is positive 

In [ ]:
n = int(input("enter pos n :"))
n1=n
sum = 0
while n > 0:
    sum+= n%10
    n=n/10
    
while n1>0:
    n1-=sum
    
print(n1)

60. Write a Python program to find the digits which are absent in a given mobile number. 

In [ ]:
n = input("numbers  :")
all_nums = set([0,1,2,3,4,5,6,7,8,9])
n = set([int(i) for i in n])
n = n.symmetric_difference(all_nums)
n = sorted(n)
print(n)

61. Write a Python program to check whether three given lengths (integers) of three sides form a right triangle. Print ”Yes” if the given sides form a right triangle otherwise print ”No”. Input: Integers separated by a single space. 1 = length of the side = 1,000 Input three integers(sides of a triangle) 8 6 7 N

In [ ]:
a=8
b=6
c=7
a*=a
b*=b
c*=c

if(a+b==c or a+c ==b or b+c ==a):
    print("yes")
else:
    print("no")

62. Writeapythonprogramtofindheightsofthetopthreebuildingindescendingorderfromeight given buildings. Input: 0 = height of building (integer) = 10,000 Input the heights of eight buildings: 25 35 15 16 30 45 37 39 Heights of the top three buildings: 45 39 37 

In [ ]:
#learnd lists from google ^^ coz we did'nt learnd it till now
print("Input the heights of eight buildings:")
l = [int(input()) for i in range(8)]
print("Heights of the top three buildings:")
l = sorted(l)
print(*l[:4:-1], sep='\n')

In [ ]:
#The End  - AhMad Kofahi